Change config api and add getClient and validateToken methods

eliasgs opened the pull request:
I had some problems with the way the constructor initialised the simple-oauth2 client (when creating a hydra clients with a configuration object the `auth.authorizePath` and `auth.tokenPath`, would default to the simple-oauth2 endpoints).

I think this way is cleaner, but I'm not sure if it breaks stuff (e.g. in the example consent app)?

Also I added a `getClient` and a `validateToken` method.

eliasgs committed:
Change how hydra clients are configured

eliasgs committed:
Add getClient method

eliasgs committed:
Add validateToken method

aeneasr commented on README.md:
```diff
@@ -36,13 +36,10 @@ $ npm i --save hydra-js
- auth: {
-   tokenHost: process.env.HYDRA_URL, // hydra url, defaults to this env var
- }
+ clientId:     process.env.HYDRA_CLIENT_ID,     // id of the client you want to use, defaults to this env var
```
Thank you for the changes! However, since this change here is only appealing to the eye I would like to keep the structure as is and prevent breaking changes! Also, it's closer to the simple-oauth2 config.

aeneasr commented on hydra.js:
```diff
@@ -88,6 +95,36 @@ class Hydra {
+          resolve(res.body)
+        })
+      }).catch((e) => {
+        console.error(e)
```
This prevents any error bubbling, instead do nothing here.

aeneasr commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hydra-js",
-  "version": "0.0.0",
+  "version": "0.1.0",
```
Please revert this change, the CI is taking care of this one :)

aeneasr reviewed (changes requested):
Awesome, thanks for the fixes. I annotated some things which need to be addressed :)

eliasgs committed:
Go back to the old config structure

This commit reverts the proposed configuration change and adds some
tests for the constructor config.

eliasgs commented:
I have made the changes you suggested. Let me know if something is still amiss:)

aeneasr reviewed (approved):
Looking good, can you confirm that this patch is working with the most recent consent-app-express master?

eliasgs commented:
As far as I can tell, the consent-app-express should work just fine (there is no breaking api changes in this PR).  I'm not sure how to easily run a complete test on the consent-app-express master though?

aeneasr commented:
Yeah that's a bit trickier, I'll check that out myself

aeneasr commented:
thanks!
